CONTRIBUTING.md

```diff
@@ -0,0 +1,79 @@
+
+
+<a name="requesting-features"></a>
+## Requesting Features
+
+To request a new feature be added to one of the existing classes, create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include:
+
+**What feature you'd like to see:**
+**Why this is important to you:** (this is here because it's interesting knowing what cool things people are working on and also could help community members make suggestions for work-arounds until the feature is built)
+
+
+
+<a name="hardware-support"></a>
+## Hardware Support
+
+Often, people get a new cool toy and wonder "does johnny five support this yet?"  When requesting support for new hardware remember that the team working on johnny-five might not own that hardware.
+
+To submit a request to support new hardware, first create a [github issue](https://github.com/rwaldron/johnny-five/issues) and include a link to the specs of the hardware and where one can purchase it. Often, it might be suggested that you be the one to help impliment the support yourself if you're already in possession of the product.
+
+<a name="pullrequests"></a>
+## Submitting Pullrequests
+
+To contribute code to Johnny Five, fork the project onto your github account and do your work in a branch. Before you submit the PR, make sure to rebase master into your branch so that you have the most recent changes and nothing breaks or conflicts.  Lint and test your code using [grunt](https://github.com/gruntjs/grunt). Also squash your commits to a reasonable size before submitting.
+
+If you are contributing code, it must include unit tests that fail if the working code isn't present and succeed when it is. Also make sure you run `grunt jsbeautifier` to fix any syntax issues.
+
+All contributions must adhere to the [Idiomatic.js Style Guide](https://github.com/rwldrn/idiomatic.js),
+by maintaining the existing coding style.
+
+
+<a name="writing-tests"></a>
+## Writing Tests
+
+Test are written using [sinon](http://sinonjs.org/).  If you are having issues making a test pass, ask for help in the Johnny-Five [gitter](https://gitter.im/) room.  Tests can be the hardest part to write when contributing code, so don't be discouraged.
+
+If you're interested in just writing tests to learn more about the project, check out the [tests](https://github.com/rwaldron/johnny-five/labels/Tests) label in the issues.
+
+<a name="writing-docs"></a>
+## Writing Documentation
+
+We are always looking to improve our docs.  If you find that any are lacking information or have wrong information, fix and submit a PR.  If you're looking for areas to start writing docs for, see the [docs](https://github.com/rwaldron/johnny-five/labels/DOCS) label in issues.
+
+Docs should have tested and working sample code, [fritzing diagrams](http://fritzing.org/), photos, and videos where applicable.  Many people using johnny five are learning how to work with hardware for the first time, so write for a beginner audience.
+
```

Comment:
Ha, I hadn't seen that. I was thinking of the actual /docs folder. It's kind of confusing. We have /eg, /docs and the wiki. /eg are the examples (and there are lots of them), /docs are annotated examples in markdown that are automatically generated from the /eg files. The actual docs are in the wiki.